src/controllers/product-controller.ts

```diff
@@ -0,0 +1,72 @@
+import {Context} from 'koa';
+import {
+    retrieveAllProducts,
+    retrieveSpecificProduct,
+    createProduct,
+    updateProduct,
+    deleteProduct,
+    checkout
+} from '../actions/product-db-actions';
+import {creationSchema, updateSchema, checkoutSchema} from '../validation-schemas/product-request-schemas';
+import {logger} from '../logger/logger';
+
+export const getAllProductsRequest = async (ctx: Context) => {
+    ctx.body = await retrieveAllProducts();
+};
+
+export const getSpecificProductRequest = async (ctx: Context) => {
+    const product = await retrieveSpecificProduct(ctx.params.id);
+    if (!product)
+        ctx.throw(404, 'Product not found!');
+    ctx.body = product;
+};
+
+export const createProductRequest = async (ctx: Context) => {
+    let validatedBody;
+    try {
+        validatedBody = await creationSchema.validateAsync(ctx.request.body);
+    } catch (err) {
+        ctx.throw(400, err);
+    }
+    const newProduct = await createProduct(validatedBody);
+    ctx.body = newProduct;
+    ctx.status = 201;
```

Comment:
https://github.com/jeffijoe/koa-respond
nice package, use it if you want (just for you to know)

Reply:
I tried that sometime but types were missing